woocommerce/woocommerce-admin
# Avoid updating customer info synchronously when "last active" is modified.

Fixes #6570.

This PR seeks to utilize action scheduler to process customer updates in the background when the `wc_last_active` user meta is updated. Prior to these changes, front end page loads (that updated the `wc_last_active` meta) would trigger a full customer sync.

The issue also called for auditing the other hooks:

> Seems like we should revisit all of the hooks being used in the customers data store and shift those tasks to jobs.

The other hooks used are `edit_user_profile_update` and `woocommerce_analytics_delete_order_stats`. The former is only fired from the backend user profile edit page, and the latter is internal to WC Admin and fires when orders are deleted (another backend - or admin-only flow).

### Detailed test instructions:

- Log in to your test store as a customer user that already exists (has a record in the Customers Report)
- Check in Tools > Scheduled Actions that a job was queued for updating that user ID





Performance: Avoid updating customer info synchronously from the front end.
